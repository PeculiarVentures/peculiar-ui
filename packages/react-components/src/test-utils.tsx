import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
