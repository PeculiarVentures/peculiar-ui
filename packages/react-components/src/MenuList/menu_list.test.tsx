import { renderWithWrapper as render } from '../test-utils';
import { MenuList } from './index';

describe('<MenuList />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(
      <MenuList>
        <li>Item-1</li>
        <li>Item-2</li>
        <li>Item-3</li>
      </MenuList>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const { asFragment } = render(
      <MenuList className="my-class-name">
        <li>Item-1</li>
        <li>Item-2</li>
        <li>Item-3</li>
      </MenuList>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
