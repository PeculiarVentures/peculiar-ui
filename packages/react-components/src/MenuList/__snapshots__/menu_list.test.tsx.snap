// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuList /> should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: var(--pv-size-base-2) 0px;
  outline: none;
  list-style: none;
  margin: 0;
}

<ul
    class="my-class-name emotion-0"
    role="menu"
    tabindex="-1"
  >
    <li>
      Item-1
    </li>
    <li>
      Item-2
    </li>
    <li>
      Item-3
    </li>
  </ul>
</DocumentFragment>
`;

exports[`<MenuList /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: var(--pv-size-base-2) 0px;
  outline: none;
  list-style: none;
  margin: 0;
}

<ul
    class="emotion-0"
    role="menu"
    tabindex="-1"
  >
    <li>
      Item-1
    </li>
    <li>
      Item-2
    </li>
    <li>
      Item-3
    </li>
  </ul>
</DocumentFragment>
`;
